<a href="https://colab.research.google.com/github/Tanu-N-Prabhu/Python/blob/master/Coffee_to_Insights.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# From Coffee to Insights: A Beginner’s Guide to Data Science

Every time I step into my favorite café, I wonder:
- Which day is the busiest for coffee sales?
- Do people eat more pastries on weekends or weekdays?
- And which day brings in the highest revenue?

Instead of guessing, let’s let data science answer these questions. In this article, I’ll walk you through a simple dataset, some Python code (Google Colab friendly), and the insights we uncover.

## The Dataset

Here’s a small Coffee Shop Sales dataset we’ll analyze:

| Date       | Day       | Coffee Sold | Pastries Sold | Revenue (\$) |
| ---------- | --------- | ----------- | ------------- | ------------ |
| 2025-08-01 | Monday    | 120         | 45            | 860          |
| 2025-08-02 | Tuesday   | 95          | 38            | 720          |
| 2025-08-03 | Wednesday | 110         | 50            | 800          |
| 2025-08-04 | Thursday  | 105         | 42            | 770          |
| 2025-08-05 | Friday    | 130         | 55            | 940          |
| 2025-08-06 | Saturday  | 85          | 70            | 780          |
| 2025-08-07 | Sunday    | 90          | 65            | 760          |


## Setting Up in Google Colab

Copy-paste this into a new Colab notebook cell:

In [1]:
import pandas as pd

# Coffee Shop Sales dataset
data = {
    "Date": ["2025-08-01", "2025-08-02", "2025-08-03", "2025-08-04", "2025-08-05", "2025-08-06", "2025-08-07"],
    "Day": ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"],
    "Coffee Sold": [120, 95, 110, 105, 130, 85, 90],
    "Pastries Sold": [45, 38, 50, 42, 55, 70, 65],
    "Revenue ($)": [860, 720, 800, 770, 940, 780, 760]
}

df = pd.DataFrame(data)
df

Date        Day  Coffee Sold  Pastries Sold  Revenue ($)
0  2025-08-01     Monday          120             45          860
1  2025-08-02    Tuesday           95             38          720
2  2025-08-03  Wednesday          110             50          800
3  2025-08-04   Thursday          105             42          770
4  2025-08-05     Friday          130             55          940
5  2025-08-06   Saturday           85             70          780
6  2025-08-07     Sunday           90             65          760

> This will give you the clean table of data inside your notebook.

## Question 1: What days are the busiest (coffee sales)?

In [2]:
busiest_day = df.loc[df["Coffee Sold"].idxmax()]
busiest_day

Date             2025-08-05
Day                  Friday
Coffee Sold             130
Pastries Sold            55
Revenue ($)             940
Name: 4, dtype: object

### Answer:
> Friday is the busiest day, with 130 cups of coffee sold.
Makes sense, the weekend is near, and people fuel up!

## Question 2: Do pastries sell more on weekends or weekdays?

In [3]:
# Calculate totals
weekday_pastries = df.loc[df["Day"].isin(["Monday","Tuesday","Wednesday","Thursday","Friday"]), "Pastries Sold"].sum()
weekend_pastries = df.loc[df["Day"].isin(["Saturday","Sunday"]), "Pastries Sold"].sum()

weekday_pastries, weekend_pastries

(np.int64(230), np.int64(135))

### Answer:
> Weekends win!

> Weekday pastries sold = 230

> Weekend pastries sold = 135
People treat themselves with pastries on weekends more than weekdays.

## Question 3: What day has the highest revenue?

In [4]:
highest_revenue = df.loc[df["Revenue ($)"].idxmax()]
highest_revenue

Date             2025-08-05
Day                  Friday
Coffee Sold             130
Pastries Sold            55
Revenue ($)             940
Name: 4, dtype: object

#### Answer:
> Friday again! With $940 revenue, it’s the most profitable day.

## Key Takeaways

- Friday is the busiest coffee day and brings the highest revenue.

- Pastries shine on weekends, customers slow down and indulge.

- Even with a tiny dataset, we can already see patterns and tell stories.


## Final Thought

This small exercise shows how data science isn’t just about numbers, it’s about uncovering insights and telling stories that matter. Whether you’re running a coffee shop or analyzing millions of rows of customer data, the principles remain the same: ask questions, explore data, and let the story unfold.